dlt-database/migrations/0001-init_db.ts

```diff
@@ -0,0 +1,130 @@
+/* FIRST MIGRATION
+ *
+ * This migration is special since it takes into account that
+ * the database can be setup already but also may not be.
+ * Therefore you will find all `CREATE TABLE` statements with
+ * a `IF NOT EXISTS`, all `INSERT` with an `IGNORE` and in the
+ * downgrade function all `DROP TABLE` with a `IF EXISTS`.
+ * This ensures compatibility for existing or non-existing
+ * databases.
+ */
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  // write upgrade logic as parameter of queryFn
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`users\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`gradido_id\` char(36) DEFAULT NULL,
+      \`derive1_pubkey\` binary(32) NOT NULL,
+      \`created_at\` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
+      \`confirmed_at\` datetime(3) DEFAULT NULL,
+      PRIMARY KEY (\`id\`),
+      UNIQUE KEY \`gradido_id\` (\`gradido_id\`),
+      UNIQUE KEY \`pubkey\` (\`pubkey\`)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;`)
+
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`accounts\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`user_id\` int(10) unsigned DEFAULT NULL,
+      \`derivation_index\` int(10) unsigned NOT NULL,
+      \`derive2_pubkey\` binary(32) NOT NULL,
+      \`type\` tinyint unsigned NOT NULL,
+      \`created_at\` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
+      \`confirmed_at\` datetime(3) DEFAULT NULL,
+      \`balance\` decimal(40,20) NOT NULL DEFAULT 0,
+      \`balance_date\` datetime(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3), 
+      PRIMARY KEY (\`id\`),
+      UNIQUE KEY \`pubkey\` (\`pubkey\`),
+      FOREIGN KEY (\`user_id\`) REFERENCES users(id)
+    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+  `)
+
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`communities\` (
```

Comment:
only a hint:
as mentioned above in the context of the users gradidoId a community will get an UUID too in the near future.
This is planed to exchange an UUID between the communities during an authentication process in an encrypted manner. And as soon as a foreign community entry will have its uuid this foreign community is marked as "authenticated", because there was a successful processed authentication between both communities.

Reply:
Yeah and I plan to use the UUID as iota_topic. Or if it is a secret, I can make a hash from it and use the hash as iota topic.